serde/src/java/org/apache/hadoop/hive/serde2/avro/SchemaToTypeInfo.java

```diff
@@ -186,6 +188,20 @@ public static TypeInfo generateTypeInfo(Schema schema,
     return typeInfoCache.retrieve(schema, seenSchemas);
   }
 
+  private static int getIntValue(JsonNode jsonNode) {
+    int value = 0;
+    if (jsonNode instanceof TextNode) {
+      try {
```

Comment:
Nit: Instead of using a try-catch block, you could use StringUtils.isNumeric() to determine if a string is a positive decimal number.

Reply:
Done.